week-1-ts-ecosystem/w01e03-object-oriented-structure/src/models/product.model.ts

```diff
@@ -0,0 +1,45 @@
+import { ProductType, UpdateProductArgs } from '../types';
+
+export class Product<Type extends ProductType> {
+  #name: string;
+  #id: string;
+  #price: number;
+  #type: Type;
+
+  constructor(name: string, id: string, price: number, type: Type) {
+    this.#name = name;
+    this.#id = id;
+    this.#price = price;
+    this.#type = type;
+  }
+
+  get name(): string {
+    return this.#name;
+  }
+
+  get id(): string {
+    return this.#id;
+  }
+
+  get price(): number {
+    return this.#price;
+  }
+
+  get type(): Type {
```

Comment:
Moim zdaniem te gettery nie mając żadnej logiki wewnątrz są nadmiarowe.
To sam TS będzie nas pilnował co do poprawności typu danych w modelu.

Natomiast fakt braku możliwości zmiany tych danych per model - może pilnować sam koszyk, jako fasada naszego dostępu do danych.